pkg/apis/istio/v1alpha2/istiocontrolplane_types.proto

```diff
@@ -256,6 +256,16 @@ message TelemetryFeatureSpec {
         string namespace = 1;
 
         TelemetryComponentSpec telemetry = 20;
+        // Optional
+        KialiComponentSpec kiali = 21;
+        // Optional
+        TracingComponentSpec tracing = 22;
+        // Optional
+        PrometheusComponentSpec prometheus = 23;
+        // Optional
+        PrometheusOperatorComponentSpec prometheusOperator = 24;
```

Comment:
prometheus operator will not work here if you see it from upstream: https://github.com/istio/installer/tree/master/istio-telemetry/prometheus-operator/templates
Since it just create prometheus operator api, we may need figure out how to integrate istio operator with these third party operators(prometheus, kiali, jaeger...)

Reply:
In cmd/mesh/manifest-generate_test.go:TestManifestGenerateFlags-all_on we expects the prometheus operator resource there.

Reply:
We already have prometheus and grafana under a third-party feature: 
https://github.com/istio/operator/blob/3a5dede6e691832e0be49e4080b62c4dba7ca9fe/pkg/component/feature/feature.go#L261
The way we support these is just in the code but not through this proto, so fields are just passed through without defining a schema of any kind. For now I suggest following the same approach, we can consider adding some kind of extensions framework in a future release.

Reply:
Currently,  ICP.RenderManifest() will loop all features in the ICP and generate manifest accordingly. Since the ThirdPartyFeature is not in ICP proto, so it will be ignored. This is also the reason why all add-on can not be installed even if you specify them in `Values` part. 
Do you suggest to put them into ThirdPartyFeature and add en entry into the ICP proto?

Reply:
There is a check in the code for ThirdParty feature so it does not need to be in the proto. I'm suggesting adding them as components in the code which will map to the correct chart directories. Then values can be passed through to the charts directly.

Reply:
added logic to check thirdparty feature from `Values` part instead of modifing the proto.

Reply:
I don't see any changes, did you push?

Reply:
No, this is not the case. with current code base, you can not have prometheus installed with default or demo profile. as described in  istio/istio#16970. Even if we have `prometheus enabled :true` in the default.yaml.

So I will use this PR to submit the fix.